### PR TITLE
Restore test_full_node_cold_sync and make it less fragile

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui/tests/full_node_tests.rs [hunk 1]
```diff
@@ -393,6 +393,36 @@ async fn test_full_node_indexes() -> Result<(), anyhow::Error> {
...
+    telemetry_subscribers::init_for_testing();
+
+    let (swarm, mut context, _) = setup_network_and_wallet().await?;
+
+    let (_, _, _, _) = transfer_coin(&mut context).await?;
+    let (_, _, _, _) = transfer_coin(&mut context).await?;
+    let (_, _, _, _) = transfer_coin(&mut context).await?;
+    let (_transfered_object, _sender, _receiver, digest) = transfer_coin(&mut context).await?;
+
+    sleep(Duration::from_millis(1000)).await;
```
Comment (longbowlu) [inline crates/sui/tests/full_node_tests.rs:408]: we should sleep after `wait_for_tx`, right?

Let's add a comment on why we sleep here

**Review by velvia — APPROVED** [review]
Wait a minute

1) I thought this test was already there?
2) The test looks identical to what I have in my branch, except that you swapped get_transactions_from_addr for `handle_transaction_info_request`..... so its because you switched it to not be based on indexing?

**Review by longbowlu — APPROVED** [review]
thanks for adding it back!

**Comment by longbowlu** [comment]
@velvia  It was accidentally deleted in my PR, probably due to rebase

File: crates/sui/tests/full_node_tests.rs (hunk 1 above)
Comment (mystenmark) [reply crates/sui/tests/full_node_tests.rs:408]: This sleep is to make sure that the authorities are quiescent before we start up the node - if they aren't, its not really a cold sync.

**Pull request merged**